teste2e/test_format_upgrades.py

```diff
@@ -0,0 +1,127 @@
+from util import UrsadbTestContext
+from util import ursadb  # noqa
+import json
+import subprocess
+import time
+from pathlib import Path
+import tempfile
+import os
+import zmq
+import json
+from typing import Dict, Any
+
+
+class TemporaryStorage:
+    def __init__(self):
+        self.tmpfiles = []
+        self.tmpdir = tempfile.gettempdir()
+
+    def tmpfile(self, name=None) -> Path:
+        if name is None:
+            filepath = self.tmpdir + "/" + os.urandom(8).hex()
+        else:
+            filepath = self.tmpdir + "/" + name
+        self.tmpfiles.append(filepath)
+        return Path(filepath)
+
+    def write_json(self, data: Dict[str, Any], name=None) -> Path:
+        p = self.tmpfile(name=name)
+        p.write_text(json.dumps(data))
+        return p
+
+    def write_text(self, text: str, name=None) -> Path:
```

Comment:
```suggestion
    def write_text(self, text: str, name: Optional[str] = None) -> Path:
```

This requires to `from typing import Optional`!